src/internal/event/EventInteractor.ts

```diff
@@ -0,0 +1,184 @@
+import { BKTException } from '../../BKTExceptions'
+import { Clock } from '../Clock'
+import { IdGenerator } from '../IdGenerator'
+import { Evaluation } from '../model/Evaluation'
+import { EventType, Event } from '../model/Event'
+import { ApiId, MetricsEventType } from '../model/MetricsEventData'
+import { User } from '../model/User'
+import { ApiClient } from '../remote/ApiClient'
+import { newBaseEvent, newDefaultEvaluationEvent, newErrorMetricsData, newEvaluationEvent, newGoalEvent, newLatencyMetricsData, newMetadata, newMetricsEvent, newSizeMetricsData } from './EventCreators'
+import { EventStorage } from './EventStorage'
+import { SendEventsFailure, SendEventsResult, SendEventsSuccess } from './SendEventResult'
+
+export class EventInteractor {
+  constructor(
+    private eventsMaxBatchQueueCount: number,
+    private apiClient: ApiClient,
+    private eventStorage: EventStorage,
+    private clock: Clock,
+    private idGenerator: IdGenerator,
+    private appVersion: string
+  ) { }
+
+  eventUpdateListener: ((events: Event[]) => void) | null = null
+
+  setEventUpdateListener(listener: ((events: Event[]) => void) | null): void {
+    this.eventUpdateListener = listener
+  }
+
+  trackEvaluationEvent(featureTag: string, user: User, evaluation: Evaluation): void {
+    this.eventStorage.add({
+      id: this.idGenerator.newId(),
+      type: EventType.EVALUATION,
+      event: newEvaluationEvent(
+        newBaseEvent(this.clock.currentTimeSeconds(), newMetadata(this.appVersion, '', '')),
+        {
+          featureId: evaluation.featureId,
+          featureVersion: evaluation.featureVersion,
+          variationId: evaluation.variationId,
+          userId: user.id,
+          user,
+          reason: evaluation.reason,
+          tag: featureTag,
+        }
+      ),
+    })
+
+    this.notifyEventsUpdated()
+  }
+
+  trackDefaultEvaluationEvent(featureTag: string, user: User, featureId: string): void {
+    this.eventStorage.add({
+      id: this.idGenerator.newId(),
+      type: EventType.EVALUATION,
+      event: newDefaultEvaluationEvent(
+        newBaseEvent(this.clock.currentTimeSeconds(), newMetadata(this.appVersion, '', '')),
+        {
+          featureId,
+          featureVersion: 0,
+          variationId: '',
+          userId: user.id,
+          user,
+          reason: {
+            type: 'DEFAULT'
```

Comment:
Please use `CLIENT` instead. The `DEFAULT` is used on the server side.

Reply:
fixed in [`7346096` (#26)](https://github.com/bucketeer-io/javascript-client-sdk/pull/26/commits/734609694f9c886db4656d14b46a5ec4c43f8cec)